c2rust-refactor/src/analysis/ownership/mod.rs

```diff
@@ -196,6 +197,46 @@ fn analyze_inter<'lty, 'tcx>(cx: &mut Ctxt<'lty, 'tcx>) {
     inter_cx.finish();
 }
 
+fn is_mut_t(ty: &TyS) -> bool {
+    if let TyKind::RawPtr(mut_ty) = ty.sty {
+        if mut_ty.mutbl == Mutability::MutMutable {
+            if let TyKind::Param(param_ty) = mut_ty.ty.sty {
+                return param_ty.name.as_str().get() == "T";
+            }
+        }
+    }
+
+    false
+}
+
+fn register_std_constraints<'a, 'tcx, 'lty>(
```

Comment:
Eventually we may want a more generic way to add constraints to lib functions, but I think we shouldn't add that now. A TODO comment mentioning that we should make it more generic if more functions are added would be nice though.